server/runtime/src/main/java/org/infinispan/server/SecurityActions.java

```diff
@@ -65,7 +65,19 @@ static void startProtocolServer(ProtocolServer<ProtocolServerConfiguration> serv
 
    static void postStartProtocolServer(Collection<ProtocolServer> servers) {
       doPrivileged(() -> {
-         servers.forEach(ProtocolServer::postStart);
+         for (ProtocolServer<?> server : servers) {
+            try {
+               server.postStart();
+            } catch (Throwable t) {
+               Server.log.protocolFailedToStart(server.getName(), t);
+
+               try {
+                  server.stop();
+               } catch (Throwable t2) {
```

Comment:
I don't think we want to start the server at all if there is a failure. I would probably not even bother stopping the protocol and exit immediately